dist/well-known.js

```diff
@@ -139,17 +139,55 @@ return Promise.all([
       if (text.startsWith('-----BEGIN PGP SIGNED MESSAGE-----')) {
         data['signed'] = true;
       }
-      for(let line of text.split('\n')) {
-        if (line.startsWith('Canonical: ')) {
-          data['canonical'] = line.substring(11);
-        } else if (line.startsWith('Encryption: ')) {
-          data['encryption'] = line.substring(12);
+      data['contact'] = [];
+      data['expires'] = [];
+      data['encryption'] = [];
+      data['acknowledgments'] = [];
+      data['preferred_languages'] = [];
+      data['canonical'] = [];
+      data['policy'] = [];
+      data['hiring'] = [];
+      data['csaf'] = [];
+      data['other'] = []; // [(name, value)]
+      for (let line of text.split('\n')) {
+        if (line.startsWith('Contact: ')) {
+          data['contact'].push(line.substring(9).trim());
         } else if (line.startsWith('Expires: ')) {
-          data['expires'] = line.substring(9);
+          data['expires'].push(line.substring(9).trim());
```

Comment:
Why do these need to be arrays instead of just string value?

Reply:
The specification states that some fields (expires and preferred-languages) are only allowed to occur once, other fields are allowed to occur several times.

By using arrays, we can easily count how often fields occur multiple times or not and for a basic validity check validate that the fields that are only allowed to occur once do not occur multiple times.
Also we are not simply choosing the first or the last one if multiple entries exist. We could also do a string concatenation in such cases, however, that would make it more difficult to split the values once again if we want to ask whether some fields occur more than once.